src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+      case 'clear':
+        for (const prop in state) {
+          if (state.hasOwnProperty(prop)) {
+            delete state[prop];
+          }
+        }
```

Comment:
I would add a blank line after each break, so it would be easier to read 🤝

```suggestion
      case 'addProperties':
        Object.assign(state, action.extraData);
        break;
        
      case 'removeProperties':
        for (const key of action.keysToRemove) {
          delete state[key];
        }
        break;
        
      case 'clear':
        for (const prop in state) {
          if (state.hasOwnProperty(prop)) {
            delete state[prop];
          }
        }
```